PhysicsTools/HepMCCandAlgos/plugins/PDFWeightsTest.cc

```diff
@@ -0,0 +1,114 @@
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "SimDataFormats/GeneratorProducts/interface/HepMCProduct.h"
+#include "SimDataFormats/GeneratorProducts/interface/LHEEventProduct.h"
+#include "SimDataFormats/GeneratorProducts/interface/GenEventInfoProduct.h"
+#include "SimGeneral/HepPDTRecord/interface/ParticleDataTable.h"
+#include "PhysicsTools/HepMCCandAlgos/interface/PDFWeightsHelper.h"
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "TTree.h"
+#include <iostream>
+using namespace std;
+using namespace edm;
+using namespace HepMC;
+
+class PDFWeightsTest : public EDAnalyzer {
```

Comment:
@bendavid - could you change this to inherit from a non-legacy module type. Thanks!